app/api/routes.py

```diff
@@ -244,7 +285,28 @@ def create_resource(json, db):
         paid=json.get('paid'),
         notes=json.get('notes'))
 
-    db.session.add(new_resource)
-    db.session.commit()
+    try:
+        db.session.add(new_resource)
+        db.session.commit()
+
+        return standardize_response(new_resource.serialize, None, "ok")
+    except Exception as e:
+        logger.error(e)
+        return standardize_response(None, [{"code": "bad-request"}], "bad request", 400)
+
 
-    return standardize_response(new_resource.serialize, None, "ok")
+def create_new_apikey(email):
+    new_key = Key(
```

Comment:
TODO: create the key in a while loop to ensure that accidental duplicates are caught and handled (in the extremely unlikely chance that there's a collision).